src/Init/System/IO.lean

```diff
@@ -67,7 +68,15 @@ constant allocprof {α : Type} (msg : @& String) (fn : IO α) : IO α := arbitra
 @[extern "lean_io_initializing"]
 constant IO.initializing : IO Bool := arbitrary _
 
-abbrev MonadIO (m : Type → Type) := HasMonadLiftT IO m
+class MonadIO (m : Type → Type) extends HasMonadLiftT IO m, MonadExcept IO.Error m
+
+instance : MonadIO IO := {}
+
+/- Omitted instances of MonadIO: OptionT, ExceptT and EStateT. The possibility for
+errors introduces the risk that `withStdout` will not restore the previous handle when
```

Comment:
Ah, I've actually removed `withStdout` etc. since it's not part of the actual implementation anymore and I didn't see any reason to keep it. Shall we still keep the changed `MonadIO`?